test/sequential/sequencerStaking.cancel.test.ts

```diff
@@ -1,403 +1,281 @@
-    await testUser1.refreshAmounts(AssetWallet.BEFORE);
-    await createAnUpdateAndCancelIt(testUser1, testUser2, chain, false);
-    await waitForNBlocks((await Rolldown.disputePeriodLength()).toNumber());
-    await testUser2.refreshAmounts(AssetWallet.AFTER);
-    await testUser1.refreshAmounts(AssetWallet.AFTER);
-    const slashRewardUpdater = testUser1
-      .getAsset(MGA_ASSET_ID)
-      ?.amountAfter.free!.sub(
-        testUser1.getAsset(MGA_ASSET_ID)?.amountBefore.free!,
-      );
-    const slashFineUpdater = testUser1
-      .getAsset(MGA_ASSET_ID)
-      ?.amountBefore.reserved!.sub(
-        testUser1.getAsset(MGA_ASSET_ID)?.amountAfter.reserved!,
-      );
-    expect(slashRewardUpdater).bnEqual(BN_ZERO);
-    expect(slashFineUpdater).bnEqual(BN_ZERO);
-  });
+it("GIVEN a sequencer, WHEN <no> canceling an update THEN no slash is applied", async () => {
+  await testUser1.refreshAmounts(AssetWallet.BEFORE);
+  await createAnUpdate(testUser1, chain);
+  await waitForNBlocks(disputePeriodLength);
+  await testUser1.refreshAmounts(AssetWallet.AFTER);
+  const updaterPenaltyValue = testUser1
+    .getAsset(GASP_ASSET_ID)
+    ?.amountBefore.reserved!.sub(
+      testUser1.getAsset(GASP_ASSET_ID)?.amountAfter.reserved!,
+    );
+  expect(updaterPenaltyValue).bnEqual(BN_ZERO);
+});
+
+it("GIVEN a slashed sequencer, WHEN slashed it can not provide any update / cancel until the next session ( if gets elected )", async () => {
+  let updaterRightsStatus: any;
+  const { reqIdCanceled, api } = await createAnUpdateAndCancelIt(
+    testUser1,
+    testUser2Address,
+    chain,
+  );
+  await waitForNBlocks(disputePeriodLength);
+  updaterRightsStatus = await Rolldown.sequencerRights(
+    chain,
+    testUser1.keyRingPair.address,
+  );
+  expect(updaterRightsStatus.cancelRights.toString()).toBe("1");
+  const txIndex = await Rolldown.lastProcessedRequestOnL2(chain);
+  //we approve the cancellation
+  await Rolldown.waitForReadRights(testUser2Address);
+  const cancelResolutionEvents = await Sudo.asSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      testUser2Address,
+      new L2Update(api)
+        .withCancelResolution(txIndex, reqIdCanceled, true)
+        .on("Ethereum")
+        .build(),
+    ),
+  );
+  await waitSudoOperationSuccess(cancelResolutionEvents, "SudoAsDone");
+  await waitForNBlocks(disputePeriodLength);
+  updaterRightsStatus = await Rolldown.sequencerRights(
+    chain,
+    testUser1.keyRingPair.address,
+  );
+  expect(updaterRightsStatus.cancelRights.toString()).toBe("0");
+});
 
-  it("GIVEN a slashed sequencer, WHEN slashed it can not provide any update / cancel until the next session ( if gets elected )", async () => {
-    const { reqIdCanceled, api } = await createAnUpdateAndCancelIt(
-      testUser1,
+it("GIVEN a sequencer, WHEN <in-correctly> canceling an update AND some pending updates/cancels, THEN it can be still slashed and kicked, cancels & updates will be executed.", async () => {
```

Comment:
looks like the test is not doing what It should be doing. The test should be about: 

1. u1 - create an update
2. u2 - Cancel it
3. u2 - create an update
4. u1 - cancel it.
5. judge tells that u2 cancel is wrong ( at step 2, with justified false ) 
6. u2 it must be kicked.
7. update 1 is executed
8. judge tells that u1 cancel is wrong. 
9. u1 must be kicked.
10. even though u2 was kicked, the update is executed.
Expected: 
The most important is to ensure that, u2 got kicked and the update was executed.
* updates for u1 and u2 are executed. ( verify the execution or execution result )
* both updaters are kicked. ( no read, no writes )

Reply:
Test was changed